# ImRep publication

Here we provide scripts and commands we used in our study "Profiling immunoglobulin repertoires across multiple human 
tissues by RNA Sequencing"  to run existing Ig and TCR repertoire assembly tools on both simulated and real data. 
The full publication is available in [Nature Communications](https://doi.org/10.1038/s41467-020-16857-7)

**Table of contents**

* [Reproducing results](#reproducing-results)
  * [Tools](#tools)
  * [Data](#data)
  * [Notebooks and Figures](#notebooks-and-figures)
* [Contact](#contact)


# Reproducing results

## Tools

We have compared ImRep to the following repertoire assembly tools:

* MIXCR (v 2.2), http://mixcr.readthedocs.io/en/latest/
* IMSEQ (v 1.1.0), http://www.imtools.org/
* IGBLAST pipeline (v 1.6.1), https://github.com/nbstrauli/influenza_vaccination_project

Additional details on how we ran these tools can be found in our supplementary 
[README](https://github.com/Mangul-Lab-USC/imrep-companion/tree/master/supplementary)

## Data

This study used both real data and simulated data. The data included in this repository is only from results of methods
that have been described here and in the [publication](https://doi.org/10.1038/s41467-020-16857-7). 

[summary_data](https://github.com/Mangul-Lab-USC/ImReP_publication/tree/master/summary_data) includes all of the data
that directly created all figures in the manuscript.

[supplementary](https://github.com/Mangul-Lab-USC/ImReP_publication/tree/master/supplementary) includes all of the 
supplementary data files attached to the manuscript.

While the raw data is not available through this repository, we are including the online sources for all raw data here:

All RNA-Seq data discussed in this paper is available as part of the Genotype-Tissue Expression (GTEx) Project under 
the phs000424.v8.p2 accession number in the database of Genotypes and Phenotypes (dbGaP). Further metadata for every 
sample can be found in our supplementary 
[data](https://github.com/Mangul-Lab-USC/imrep-companion/tree/master/supplementary/Supplementary_Data_4.csv)

The targeted BCR-Seq assemblies used for validation of the ImReP method are available at the adaptive biosystems 
webpage (https://clients.adaptivebiotech.com/pub/lombardo-2017-bloodadvances). 

RNA-Seq samples used for validation of ImReP are available under the SRP099346 accession number in the 
Sequence Read Archive. 

Raw sequence reads of targeted BCR-Seq data are available under the EGAN00001419382 accession number in the 
European Genome-Phenome Archive.

An explanation of how we produced simulated data can be found in our supplementary 
[README](https://github.com/Mangul-Lab-USC/imrep-companion/tree/master/supplementary)

Some files were too large to fit in a GitHub repository, so these are being hosted by 
[figshare](https://doi.org/10.6084/m9.figshare.12210266)
Notably, this includes the data required to generate Figure 5a, as well as the zipfile containing all of the data
required to create all figures in this repository.

## Notebooks and Figures

[notebooks](https://github.com/Mangul-Lab-USC/ImReP_publication/tree/master/notebooks) contains a notebook for every
figure in the manuscript, in addition to one notebook for all supplementary figures. These notebooks produce the
panels as they appear in the manuscript.

[figures](https://github.com/Mangul-Lab-USC/ImReP_publication/tree/master/figures) contains all of the image files for 
the plots generated by all notebooks. 


# Contact

Please do not hesitate to contact us (mangul@usc.edu) if you have any comments, suggestions, or clarification requests 
regarding the study or if you would like to contribute to this resource.



